pkg/registry/node/etcd/etcd.go

```diff
@@ -102,3 +103,25 @@ var _ = rest.Redirector(&REST{})
 func (r *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.RoundTripper, error) {
 	return node.ResourceLocation(r, r.connection, r.proxyTransport, ctx, id)
 }
+
+// An interface for types that allow getting information about a Node on which give pod is running.
+type HostLocator interface {
```

Comment:
It's not that easy. Dependencies are: master depends on KubeletClient and registry, *new* KubeletClient depends on registry, but also registry depends on KubeletClient (as e.g. whole proxy logic is there). Logical place for KubeletClient would be, well, Kubelet (probably), but we still need to untangle cyclic dependency between KubeletClient and NodeRegisty. To do so we can have this interface and make NodeREST implement it, so it can be injected during creation. @liggitt - how does it sound to you?